gradle-plugins/model-compiler/src/main/java/io/spine/gradle/compiler/model/ProcessingStages.java

```diff
@@ -0,0 +1,199 @@
+ */
+
+package io.spine.gradle.compiler.model;
+
+import com.google.common.base.Function;
+import io.spine.gradle.ProjectHierarchy;
+import io.spine.server.aggregate.Aggregate;
+import io.spine.server.command.CommandHandler;
+import io.spine.server.model.Model;
+import io.spine.server.procman.ProcessManager;
+import io.spine.tools.model.SpineModel;
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collection;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Collections2.transform;
+import static com.google.common.collect.Lists.newLinkedList;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+import static java.lang.String.format;
+import static java.util.Arrays.deepToString;
+
+/**
+ * The factory for the {@link ProcessingStage} instances.
+ *
+ * @author Dmytro Dashenkov
+ */
+final class ProcessingStages {
+
+    private static final Model model = Model.getInstance();
+
+    private ProcessingStages() {
+        // Prevent utility class instantiation.
+    }
+
+    /**
+     * Retrieves the {@link ProcessingStage} which validates the given {@link SpineModel}.
+     *
+     * <p>The validation includes the command handler checks.
+     *
+     * @param project the Gradle project to get the compiled sources from
+     */
+    static ProcessingStage validate(Project project) {
+        return new ValidatingProcessingStage(project);
+    }
+
+    /**
+     * The validating {@link ProcessingStage}.
+     *
+     * @see ProcessingStages#validate(Project)
+     */
+    private static class ValidatingProcessingStage implements ProcessingStage {
+
+        private static final URL[] EMPTY_URL_ARRAY = new URL[0];
+
+        private final URLClassLoader projectClassLoader;
+
+        private ValidatingProcessingStage(Project project) {
+            this.projectClassLoader = createClassLoaderForProject(project);
+        }
+
+        @SuppressWarnings("IfStatementWithTooManyBranches") // OK in this case.
+        @Override
+        public void process(SpineModel rawModel) {
+            for (String commandHandlingClass : rawModel.getCommandHandlingTypesList()) {
+                final Class<?> cls;
+                try {
+                    log().debug("Trying to load class \'{}\'", commandHandlingClass);
+                    cls = getModelClass(commandHandlingClass);
+                } catch (ClassNotFoundException e) {
+                    throw new IllegalStateException(e);
+                }
+                if (Aggregate.class.isAssignableFrom(cls)) {
+                    @SuppressWarnings("unchecked") final Class<? extends Aggregate> aggregateClass =
```

Comment:
Please have a single `@SuppressWarnings` for the method and a valid explanation.